### PR TITLE
feat: Remove non-core dependencies to extras section

### PR DESCRIPTION
**Pull request by joloppo**
## CHANGELOG

- Moved the packages jupyter, matplotlib and statsmodels to the extras section of the setup. 

These are not needed for the core sdk, and are only used for the demo notebooks. They bloat package install by >50MB.

## TESTS

- Testing currently fails on one test, probably unrelated. Please advise.
image

## COMMENTS

- Please advise on whether this makes sense and release process. 
- Related ticket: https://vortexa.atlassian.net/browse/RND-3595
- Thread: https://vortexa.slack.com/archives/CE4C35PN3/p1694595674073659

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -26,11 +26,9 @@
     python_requires=">=3.7",
     install_requires=[
         "flatten-dict==0.4.2",
-        "jupyter==1.0.0",
         "matplotlib>=3.5.3",
```
Comment (kujon) [inline setup.py:29]: Why is `matplotlib` required?

Comment (valeriiashmyhlo) [reply setup.py]: Tests were failing, fixed it. `matplotlib` is optional now.

File: .circleci/config.yml [hunk 2]
```diff
@@ -131,6 +131,8 @@ jobs:
             pip install -U pip
             pip install -e .
```
Comment (joloppo) [inline .circleci/config.yml:132]: This works, suggestion:

Use `pip install -e .[binder]` and add a "binder" section to the extras in `setup.py` much like the current `tests` section. Example in line 18 of this file where tests requirements are installed. you can remove all 3 (L 133, 134, 145) pip installs here after moving the requirements to the extra section.

File: setup.py [hunk 3]
```diff
@@ -26,11 +26,8 @@
     python_requires=">=3.7",
     install_requires=[
         "flatten-dict==0.4.2",
-        "jupyter==1.0.0",
-        "matplotlib>=3.5.3",
         "pandas>=0.25.2",
         "requests==2.31.0",
-        "statsmodels==0.13.2",
         "tqdm==4.64.1",
         "pydantic==1.10.2",
```
Comment (joloppo) [inline setup.py:32]: Not sure where we are using pydantic but binding it to a specific version is a bad idea.

File: setup.py [hunk 4]
```diff
@@ -26,11 +26,8 @@
     python_requires=">=3.7",
     install_requires=[
         "flatten-dict==0.4.2",
-        "jupyter==1.0.0",
-        "matplotlib>=3.5.3",
         "pandas>=0.25.2",
         "requests==2.31.0",
-        "statsmodels==0.13.2",
         "tqdm==4.64.1",
```
Comment (joloppo) [inline setup.py:31]: Not sure how we use tqdm but this could maybe be an extra as well.

**Review by valeriiashmyhlo — APPROVED** [review]

**Pull request merged**